### DISCUSSION TITLE
[ZK Stack] - <Save my deposit coins stuck in bridge!!>

### DISCUSSION THREAD

**Discussion by qiwenzhou** [zkSync-Community-Hub/zksync-developers#623] [ZK Stack] [answered]
### Module Affected

- [X] System contracts (era-system-contracts)
- [X] Smart contracts (era-contracts)
- [X] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

1.74.0

### Operating System

- [ ] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

I tried to run a hyperchain on eth and accidently deleted my db. Now I lost all my L2 data except for rocksdb-related(stateKeeper and so on).  I can run the server up but 
The account balance is all 0 and my money stuck in bridge. How to withdraw my money?

### Expected Behavior

Withdraw my money

### Current Behavior

Cannot withdraw.

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
It seems like you're facing a complex issue involving the loss of L2 data and funds stuck in the bridge after accidentally deleting your database. Here are a few steps you can take to address this problem:

1. **Restore from Backup:**
   - If you have backups of your database, the first step would be to restore from the latest backup before the deletion occurred. This could potentially recover your account balances and other critical data.

2. **Check Transaction History:**
   - If you have transaction hashes or any other identifiers related to the funds that are stuck, you can use these to trace the transactions on the blockchain. This might provide insights into the status of these funds.

3. **Contact Support:**
   - Since this issue involves system contracts and the zkEVM, it might require direct intervention from the support team. You can reach out to them with detailed information about your issue, including any relevant transaction hashes and the steps you've already taken.

4. **Bridge Interaction:**
   - If your funds are stuck in a bridge, and you have access to the transaction details, you might need to contact the bridge's support team. Provide them with the transaction hash and any other relevant details to help them understand and potentially resolve the issue.

5. **Rebuild the Node:**
   - Since you mentioned that you can run the server but the account balance shows as zero, it might be necessary to re-sync or rebuild your node entirely. This could help in recovering some of the lost data if the state is still intact in other parts of the system.

6. **Consult the Community:**
   - Sometimes, community forums or discussions (like those on GitHub or other developer platforms) might have encountered similar issues. It could be helpful to search there or post your problem to see if others can offer solutions based on their experiences.

If you need further assistance or have specific transaction hashes that you think are relevant, please provide them, and I can look into more detailed solutions or direct you to the appropriate support channels.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by bxpana** [reply]
I'll check on this, but I'm not sure you can without that data. This is related to this issue right? https://github.com/zkSync-Community-Hub/zksync-developers/discussions/609

**Reply by bxpana** [reply, marked as answer]
To my understanding, you need a fully functioning L2 node in order to withdraw funds (i.e., a node that can process L2 withdraw transactions), and this means restoring Postgres data. So you won't be able to retrieve those funds in the bridge without that data.
